### ISSUE TITLE
ParticleHandle.pos should handle coordinates outside box

### ISSUE THREAD

**Issue by RudolfWeeber** [espressomd/espresso#4324]
It is possibel to set particles outside the primary simulatino box using
```
p.pos = [-0.1]*3
```
However, the result will be different from a particle that has crossed theboundary during propagation. This is confusing

* When a particle crosses during propagation, `p.r.p` contains a folded position and `p.l.i` the corresponding image box count (such as `[-1,-1,-1]`)
* The freshly put particle has an unfolded coordinate in `p.r.p` (i.e.<0) and an image box of 0.

I think, the coordinate should be folded (and the image box count changed accordingly) on using the position setter from Python.

**Comment by jngrad** [comment]
This should take periodicity into account. For example with Stokesian Dynamics, there is no periodicity *and* no boundary conditions, i.e. it is perfectly normal for particles to move outside the simulation box.

**Comment by jngrad** [comment]
> The freshly put particle has an unfolded coordinate in p.r.p (i.e.<0) and an image box of 0.

I don't think this is true. Particle coordinates are folded when we add a particle or update an existing one:
https://github.com/espressomd/espresso/blob/72a9a17df0309a8e887fa8beecbb8cf6b182339d/src/core/particle_data.cpp#L620-L623

This is also consistent with what happens during integration:
```python
import espressomd
import numpy as np
system = espressomd.System(box_l=[1, 1, 1])
system.time_step = 0.01
system.cell_system.skin = 0.25
system.part.add(pos=[-0.1, 0, 0], v=[-0.1, 0, 0])
system.part.add(pos=[+0.1, 0, 0], v=[-0.1, 0, 0])
print(np.around(system.part[:].pos, 2))
system.integrator.run(100)
print(np.around(system.part[:].pos, 2))
system.integrator.run(100)
print(np.around(system.part[:].pos, 2))
system.integrator.run(1000)
print(np.around(system.part[:].pos, 2))
```
```
$ ./pypresso mwe4.py
[[-0.1  0.   0. ]
 [ 0.1  0.   0. ]]
[[-0.2  0.   0. ]
 [-0.   0.   0. ]]
[[-0.3  0.   0. ]
 [-0.1  0.   0. ]]
[[-1.3  0.   0. ]
 [-1.1  0.   0. ]]
```

The folded coordinates are obtained via `system.part[:].pos_folded`. When working with Lees-Edwards boundary conditions (LEbc), one should refrain from using the  `system.part[:].pos` getter. When using the position setter with LEbc, one should also refrain from passing coordinates outside the box, because one of the 3 coordinates will be folded with a large offset.
